File: Banking-Kata/src/main/java/banking/model/Account.java
package banking.model;

import java.math.BigDecimal;
import java.util.UUID;

public class Account implements Identifiable {
  
  private final UUID id;
  private Money balance;
  
  private Account(UUID id, Money balance) {
    this.id = id;
    this.balance = balance;
  }

  public Account(Money startingBalance) {
    this(UUID.randomUUID(), startingBalance);
  }
  
  public Account() {
    this(new Money(BigDecimal.ZERO));
  }

  @Override
  public UUID getId() {
    return id;
  }

  public Money getBalance() {
    return balance;
  }

  public void deposit(Money amount) {
    balance = balance.plus(amount);
  }

  public void withdraw(Money amount) {
    balance = balance.minus(amount);
  }

  public Account cloneWithBalance(Money balance) {
    return new Account(this.id, balance);
  }

}

File: 07-Banking-Kata/csharp-dotnetcore2/BankingKataAPI/Models/Account.cs
using System;
using Newtonsoft.Json;

namespace BankingKataAPI.Models
{
    public class Account : Identifiable
    {
        public Guid Id { get; }
        public Money Balance { get; }

        [JsonConstructor]
        private Account(Guid id, Money balance)
		{
			Id = id == Guid.Empty ? Guid.NewGuid() : id;
			Balance = balance;
		}

		public Account(Money balance)
            : this(Guid.Empty, balance)
		{
            
		}

        public Guid GetId()
        {
            return Id;
        }

		public Account CloneWithBalance(Money balance)
		{
			return new Account(Id, balance);
		}
    }
}
